scripts/build_dp_consensus.py

```diff
@@ -0,0 +1,154 @@
+import yaml
+from pathlib import Path
+from typing import Union
+
+import pandas as pd
+from pycytominer.operations import get_na_columns
+from pycytominer import consensus
+
+
+def infer_dp_features(dp_profile) -> list[str]:
+    """Returns a list of Deep profiler features found within the single cell
+    dataframe
+
+    Parameters
+    ----------
+    dp_profile : pd.DataFrame
+        dataframe features captured from deep profiler
+
+    Returns
+    -------
+    list[str]
+        list of deep profiler features
+
+    Raises
+    ------
+    ValueError
+        Raised if no Deep profiler features are found within the given DataFrame
+    """
+    dp_features = []
+    metadata_model = dp_profile["Metadata_Model"]
```

Comment:
oh, if this is a column, won't the next line error?

Reply:
sorry, i mean line 32 (2 lines from here)

Reply:
Hm... let me run some tests and I will get back to you on that.

Reply:
I ran some tests and it did not raise any errors. 

They `any()` if block checks if the current column contains any of the metadata_model. So it actually iterates through the `metadata_model` list.

Below is an example assuming that there are 2 different metadata models used 
```python
metadata_model = ["model", "simple"]
columns = ["location", "number", "treee", "model_1", "model_2", "model_3", "model_4", "simple_1", "simple_2"]

dp_features = []
for column in columns:
        if any([column.startswith(f"{meta_data}_") for  meta_data in metadata_model]):
            dp_features.append(column)


if len(dp_features) <= 0:
    raise ValueError(
        "No DP features found, Are you sure that this dataframe is from DeepProfiler?"
    )
```

In this test, the returning list should be everything within the `columns` but the first three.
``` python
print(dp_features)
```

and I get: 
```text
['model_1', 'model_2', 'model_3', 'model_4', 'simple_1', 'simple_2']
```

Maybe I am not seeing what you are seeing that can cause this to fail. 

However, I did fix a small bug: 38c642e0385740eb277bf444b59789a921d5093d

Reply:
Thanks for this explanation! I see now that my misread was in the list comprehension. Thanks for pointing this out!